include/mxnet/ndarray.h

```diff
@@ -766,36 +763,36 @@ class NDArray {
   /*
    * This function returns dnnl::memory with the default primitive_desc.
    */
-  const dnnl::memory* GetDNNLData() const;
+  const void* GetDNNLData() const;
```

Comment:
I agree with @PawelGlomski-Intel  that we can use _dnnl::memory*_ here and below:
```suggestion
  const dnnl::memory* GetDNNLData() const;
```
with forward declaration near line 77
```
namespace dnnl{
  struct memory;
}
```
This way we can avoid most of the static_cast in this change.